pkg/controller/migplan/close.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2019 Red Hat Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package migplan
+
+import (
+	"context"
+
+	migapi "github.com/fusor/mig-controller/pkg/apis/migration/v1alpha1"
+)
+
+// Ensure that resources managed by the MigPlan have been cleaned up
+func (r ReconcileMigPlan) ensureClosed(plan *migapi.MigPlan) error {
+	// Migration Registry
+	err := r.ensureMigRegistriesDelete(plan)
+	if err != nil {
+		return err
+	}
+
+	plan.Status.SetCondition(migapi.Condition{
+		Type:     Closed,
+		Status:   True,
+		Category: Critical,
+		Message:  ClosedMessage,
+	})
+	// Apply changes.
+	plan.Status.EndStagingConditions()
+	err = r.Update(context.TODO(), plan)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Remove the closed condition
+func (r ReconcileMigPlan) deleteClosedCondition(plan *migapi.MigPlan) error {
```

Comment:
I think something like `planReopened()` would be more symetrical with `ensureClosed()`.  Although the behavior would (initially) be to remove the `Closed` condition, the more generic name would support anything related to _un-closing_ the plan.

Reply:
My thinking wasn't so much keeping to the same go file but that we're triggering the condition changes from the same place. This is called from the `else` clause of the `if` statement that calls ensureClosed on the other side. Seemed cleaner to keep it together "if closed, close things, else open them"